### PR TITLE
test(data-pipeline): add unit tests

### PR DESCRIPTION
**Pull request by giovanni-vaccarino**
### Description

This PR introduces unit testing for the data pipeline part. In particular it includes the unit test for the following packages:
- _embedding/_
- _vectorstore/_
- _retrieve/_
- _chunking/_

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did

### CODE REVIEW

File: chatbot-core/tests/unit/rag/retrieve/test_retriever_utils.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    query_vector = np.array([0.1, 0.2], dtype=np.float32)
+
+    data, scores = search_index(
+        query_vector=query_vector,
+        index=index,
+        metadata=metadata,
+        logger=mock_logger,
+        top_k=2
+    )
+
```
Comment (cnu1812) [inline chatbot-core/tests/unit/rag/retrieve/test_retriever_utils.py:111]: Add test for metadata length mismatch. the above thet index out of bounds, is fine but you don’t test what happens if metadata is shorter than index.ntotal, which could silently fail or cause missing docs.

File: chatbot-core/tests/unit/rag/embedding/test_embed_chunks.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    mocker.patch("builtins.open", mock_open)
+    mocker.patch("json.load", side_effect=json.JSONDecodeError("Expecting value", "doc", 0))
+
+    result = load_chunks_from_file("invalid.json", mock_logger)
+
+    assert result == []
+    mock_logger.error.assert_called_once()
+    assert "JSON decode error" in mock_logger.error.call_args[0][0]
+
+
```
Comment (cnu1812) [inline chatbot-core/tests/unit/rag/embedding/test_embed_chunks.py:127]: Add a negative test where all files are either empty or contain only invalid chunks.

**Review by berviantoleo — APPROVED** [review]

**Review by cnu1812 — APPROVED** [review]

**Review by krisstern — APPROVED** [review]
